.github/workflows/ci.yaml

```diff
@@ -0,0 +1,59 @@
+name: Run tests
+
+on:
+  pull_request:
+  push:
+    branches:
+     - main
+     - feat/**
+  schedule:
+    - cron: "53 0 * * *" # Daily at 00:53 UTC
+  workflow_dispatch:
+
+jobs:
+  pre-commit:
+    name: Run pre-commits
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Install tox & poetry
+        run: |
+          pipx install tox
+          pipx install poetry
+      - name: Set up python environment
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+          cache: 'poetry'
+      - name: Install pre-commit
+        run: |
+          poetry install
+          poetry run pre-commit install
+      - name: Run pre-commit hooks
+        run: |
+          poetry run pre-commit run --all-files
+
+  lint:
+    name: Lint
+    uses: canonical/data-platform-workflows/.github/workflows/lint.yaml@v21.0.0
+
+  unit-test:
+    name: Unit test charm
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Install tox & poetry
+        run: |
+          pipx install tox
+          pipx install poetry
+      - name: Set up python environment
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.10"
+          cache: 'poetry'
+      - name: Run tests
+        run: tox run -e unit
```

Comment:
This will be done, I just didn't have time to do it yet and I need to balance between the tests that will be run in this repository and the tests run in the 4 charms repositories